aten/src/ATen/native/TensorShape.cpp

```diff
@@ -87,6 +89,19 @@ Tensor& set_cpu_(Tensor& result) {
   return result;
 }
 
+// We can't re-use the cpu kernel here because we don't want to use the cpu allocator.
```

Comment:
just tried it haha - too bad but it looks like that breaks. The codegen that spits out the native function declarations is invalid, since it emits separate function declaration / definitions in a `.h` and `.cpp` file (and templates need to always go in the header).
```
// set_native.h
namespace at {
namespace native {

// Not a valid declaration 
TORCH_API at::Tensor & set_device_type_<kCPU>(at::Tensor & self);

} // namespace native
} // namespace at
```

Reply:
Ho well, it was close! Thanks for checking.